### PR TITLE
Feature: adding method to check for orders completed

### PR DESCRIPTION
**Pull request by eapg**
* Adding method to check for orders complete
* Adding unit and integration test for this method

This method will take the order in turn from in process queue and will ask if the time.now is higher than the time when the order status chance to in process + order estimated time if the condition is true means that the passed the necessary time for the order to be complete.

### CODE REVIEW

File: src/core/engine/processors/kitchen_simulator.py [hunk 1]
```diff
@@ -57,6 +59,26 @@ def assign_orders_to_available_chefs(self):
...
+                order_id_to_be_check
+            )
+
+            if datetime.now() > (
+                last_order_status_history.from_time
+                + timedelta(seconds=order_to_be_check.estimated_time)
+            ):
+                self._order_send_to_complete(order_to_be_check)
+            else:
+                self.order_manager.add_to_queue(order_to_be_check)
```
Comment (reynoldmorel) [inline src/core/engine/processors/kitchen_simulator.py:80]: How do you know which queue to use?

File: src/core/engine/processors/kitchen_simulator.py [hunk 2]
```diff
@@ -57,6 +59,26 @@ def assign_orders_to_available_chefs(self):
...
+            OrderStatus.IN_PROCESS
+        )
+
+        if order_id_to_be_check:
+            order_to_be_check = self.order_controller.get_by_id(order_id_to_be_check)
+            last_order_status_history = self.order_status_history_controller.get_last_status_history_by_order_id(
+                order_id_to_be_check
+            )
+
+            if datetime.now() > (
```
Comment (reynoldmorel) [inline src/core/engine/processors/kitchen_simulator.py:74]: I would prefer if you convert datetime to milliseconds and then compare in milliseconds instead

File: src/tests/utils/fixtures/order_fixture.py [hunk 3]
```diff
@@ -2,12 +2,13 @@
...
 
 
-def build_order(order_id=None, status=None, assigned_chef_id=None):
+def build_order(order_id=None, status=None, assigned_chef_id=None, estimate_time=None):
 
     order = Order()
     order.id = order_id
     order.status = status or OrderStatus.NEW_ORDER
     order.assigned_chef_id = assigned_chef_id
+    order.estimated_time = estimate_time
```
Comment (reynoldmorel) [inline src/tests/utils/fixtures/order_fixture.py:11]: Why are you adding this here?

Comment (reynoldmorel) [reply src/tests/utils/fixtures/order_fixture.py:11]: Compute estimated time every time a order needs to evaluated.

**Review by reynoldmorel — CHANGES_REQUESTED** [review]

File: src/core/engine/processors/kitchen_simulator.py (hunk 2 above)
Comment (eapg) [reply src/core/engine/processors/kitchen_simulator.py:74]: done.

File: src/tests/utils/fixtures/order_fixture.py (hunk 3 above)
Comment (eapg) [reply src/tests/utils/fixtures/order_fixture.py]: done.

File: src/core/engine/processors/kitchen_simulator.py [hunk 4]
```diff
@@ -57,18 +60,43 @@ def assign_orders_to_available_chefs(self):
...
+            chef_assigned = self.chef_controller.get_by_id(
+                order_to_be_checked.assigned_chef_id
+            )
+            order_ingredients = self.order_controller.get_order_ingredients_by_order_id(
+                order_id_to_be_checked
+            )
+            order_estimate_time = compute_order_estimated_time(
+                order_ingredients, chef_assigned
+            )
+            if get_unix_time_stamp_milliseconds(
```
Comment (reynoldmorel) [inline src/core/engine/processors/kitchen_simulator.py:85]: Could store results in variables before doing the check with the if?

**Review by reynoldmorel — APPROVED** [review]

**Pull request merged**